ZFSin/spl/module/spl/spl-vnode.c

```diff
@@ -1076,16 +1076,25 @@ FILE_OBJECT *vnode_fileobject(vnode_t *vp)
 }
 
 void vnode_couplefileobject(vnode_t *vp, FILE_OBJECT *fileobject) {
+	KIRQL OldIrql;
 	
 	if (fileobject)
 		fileobject->FsContext = vp;
-
-	vnode_setfileobject(vp, fileobject);
+	if (vp) {
```

Comment:
Similarly here, although, I think it would be acceptable to require vp to already be held when calling `vnode_couplefileobject()` ? It really should already be, one hopes :)